client/pkg/omni/resources/siderolink/data/joinconfig.yaml

```diff
@@ -0,0 +1,13 @@
+---
+apiVersion: v1alpha1
+kind: SideroLinkConfig
+apiUrl: '{{ .APIURL }}'
+---
+apiVersion: v1alpha1
+kind: EventSinkConfig
+endpoint: '{{ .EventSinkEndpoint }}'
+---
+apiVersion: v1alpha1
+kind: KmsgLogConfig
+name: omni-kmsg
+url: '{{ .LogEndpoint }}'
```

Comment:
this also works, but you can generate it programmatically as well to avoid drift and issues with YAML escaping

e.g. https://github.com/siderolabs/talos/blob/519a48302e771fd9b331913166d55c50fff4961a/pkg/machinery/config/types/siderolink/siderolink_test.go#L39-L40

if you combine them into a config Container and then encode the whole, you should always get a valid machine config